### DISCUSSION TITLE
Embed PDF images

### DISCUSSION THREAD

**Discussion by Compizfox** [marp-team/marp#322] [Q&A] [answered]
Hi,

I just started out with Marp to see if it's a better solution from making slides than LaTeX Beamer.

From Beamer, I'm used to include (vector) images in PDF format. I tried to do the same with Marp:

    ![bg](example.pdf)

but the image doesn't show up after compiling with `marp presentation.md --pdf --allow-local-files`. I also do not get any error, but the image just isn't rendered.

The [documentation](https://marpit.marp.app/image-syntax) doesn't mention anything at all about supported file formats for images.

It does work with SVG images.

Am I missing something, or is this just not supported? If so, I think the supported image formats should be mentioned in the documentation, and `marp-cli` should output an error when a non-supported image is encountered, I think.

**Comment by yhatt** [comment, marked as answer]
Marp is running on Web-based technology unlike TeX based tools. `![]()` image syntax is corresponding to `<img src="">` HTML tag, and it's only working with images compatible with Web (See [Image file type and format guide on MDN](https://developer.mozilla.org/docs/Web/Media/Formats/Image_types)). 

Generally PDF is not regarded as an image file. There is a MIME type `application/pdf` in PDF, not `image/*`. To embed PDF document, use `<iframe>` tag with enabling HTML rendering by Marp CLI's `--html` option.

```html
# Embed PDF

<iframe src="file.pdf" width="100%" height="500" frameborder="0" />
```

```console
marp your-markdown.md --html --pdf --allow-local-files
```

https://forum.webdeveloper.com/d/146790-how-to-display-pdf-picture-in-html

**Reply by yhatt** [reply]
> If so, I think the supported image formats should be mentioned in the documentation, and marp-cli should output an error when a non-supported image is encountered, I think.

Marp has only concerned with Markdown conversion process. We are not making clear supported file types because [we cannot know whether PDF file is renderable with `<img>` until the output HTML renders in the browser](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#:~:text=The%20HTML%20standard%20doesn%27t%20list%20what%20image%20formats%20to%20support%2C%20so%20user%20agents%20may%20support%20different%20formats.). Errors/warnings about file type are also not showing in Marp CLI for the same reason.

For example, `![](xxx.pdf)` is actually working with Safari.

**Reply by Compizfox** [reply]
> Marp is running on Web-based technology unlike TeX based tools. ![]() image syntax is corresponding to <img src=""> HTML tag, and it's only working with images compatible with Web (See [Image file type and format guide on MDN](https://developer.mozilla.org/docs/Web/Media/Formats/Image_types)).

Thanks, I see.

> Generally PDF is not regarded as an image file. There is a MIME type application/pdf in PDF, not image/*. To embed PDF document, use <iframe> tag with enabling HTML rendering by Marp CLI's --html option.

That's not exactly what I wanted. This results in the browser's PDF viewer in an iframe, whereas I was just wondering if you could embed images in PDF format (like images in SVG format) in the presentation slide like I'm used to with Beamer.

> Marp has only concerned with Markdown conversion process. We are not making clear supported file types because [we cannot know whether PDF file is renderable with <img> until the output HTML renders in the browser](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#:~:text=The%20HTML%20standard%20doesn%27t%20list%20what%20image%20formats%20to%20support%2C%20so%20user%20agents%20may%20support%20different%20formats.). Errors/warnings about file type are also not showing in Marp CLI for the same reason.

I was using Marp to render to PDF, so I wasn't using a browser, hence my confusion ;)

I guess I'll just have to convert the images to SVG then. Not a big deal of course.

**Reply by YanchunHe** [reply]
Hi @yhatt 
Is it possible to use Safari to render the md and convert to pdf?

> For example, `![](xxx.pdf)` is actually working with Safari.

Thanks a lot!

**Reply by yhatt** [reply]
No.

> You can try [File] -> [Export as PDF] or [File] -> [Print] -> [PDF] -> [Save as PDF] in the exported HTML, _just_ for exporting PDF using Safari.

**Reply by YanchunHe** [reply]
Thanks for the reply! Yes, but this can export only one slide, instead of all slides as a whole pdf.
